src/Compilers/Core/Portable/CodeGen/ILBuilder.cs

```diff
@@ -75,8 +76,11 @@ internal ILBuilder(ITokenDeferral module, LocalSlotManager localSlotManager, Opt
 
             _labelInfos = new SmallDictionary<object, LabelInfo>(ReferenceEqualityComparer.Instance);
             _optimizations = optimizations;
+            _localsAreZeroed = localsAreZeroed;
         }
 
+        public bool LocalsAreZeroed => _localsAreZeroed;
```

Comment:
>public bool LocalsAreZeroed => _localsAreZeroed; [](start = 8, length = 48)

It looks like it is not necessary to have this property on ILBuilder or have it inside ILBuilder because ILBuilder itself doen't take any advantage of it.

Reply:
Never mind, it looks like ILBuilderToString is using this property.

---
In reply to: [167670235](https://github.com/dotnet/roslyn/pull/24723#discussion_r167670235) [](ancestors = 167670235)

Reply:
We're using it in ILBuilderVisualizer. We need this property when we call DumpMethod.